contracts/UbeswapOrderBook.sol

```diff
@@ -21,37 +22,29 @@ contract UbeswapOrderBook is OrderBook, Ownable {
     /// @notice Fee for broadcasting an order. In units of PCT_DENOMINATOR
     uint256 public fee;
 
-    /// @notice Currency which subsidies are paid out in
-    IERC20 public subsidyCurrency;
-    /// @notice Mapping of each makerToken's subsidy rate
-    mapping(address => uint256) public subsidyRate;
-
     /// @notice Fee recipient
     address public feeRecipient;
 
+    /// @notice Reward distributor module
+    IOrderRewardDistributor public rewardDistributor;
+
     event FeeChanged(uint256 oldFee, uint256 newFee);
     event FeeRecipientChanged(address oldFeeRecipient, address newFeeRecipient);
-    event SubsidyCurrencyChanged(
-        address oldSubsidyCurrency,
-        address newSubsidyCurrency
-    );
-    event SubsidyRateChanged(
-        address token,
-        uint256 oldSubsidyRate,
-        uint256 newSubsidyRate
+    event RewardDistributorChanged(
+        address oldRewardDistributor,
+        address newRewardDistributor
     );
-    event ERC20Rescued(address token, uint256 amount);
 
     constructor(
         LimitOrderProtocol _limitOrderProtocol,
         uint256 _fee,
         address _feeRecipient,
-        IERC20 _subsidyCurrency
+        IOrderRewardDistributor _rewardDistributor
     ) OrderBook(_limitOrderProtocol) {
         require(_fee <= MAX_FEE, "UOB: Fee exceeds MAX_FEE");
         fee = _fee;
         feeRecipient = _feeRecipient;
-        subsidyCurrency = _subsidyCurrency;
+        rewardDistributor = _rewardDistributor;
```

Comment:
we shouldn't have to tell the order book which reward distributor to use, it should be definable by the UI that's submitting the order as part of interaction data to fully de-couple the reward logic from the order book itself